examples/Extensions/AddSitelinksUsingFeeds.cs

```diff
@@ -0,0 +1,366 @@
+                
+                // Create a campaign feed, which tells Google Ads which campaigns to use the
+                // provided data with.
+                
+                CreateCampaignFeed(client, customerId, campaignId, feedData);
+                // If an ad group is specified, limit targeting only to the given ad group.
+                if (adGroupId.HasValue)
+                {
+                    CreateAdGroupTargeting(client, customerId, feedData, adGroupId.Value);
+                }
+            }
+            catch (GoogleAdsException e)
+            {
+                Console.WriteLine("Failure:");
+                Console.WriteLine($"Message: {e.Message}");
+                Console.WriteLine($"Failure: {e.Failure}");
+                Console.WriteLine($"Request ID: {e.RequestId}");
+                throw;
+            }
+        }
+
+        private Hashtable CreateFeed(GoogleAdsClient client, long customerId)
+        {
+            Feed feed = new Feed()
+            {
+                Name = $"Sitelinks Feed {DateTime.Now:yyyy-MM-dd hh:mm:ss}",
+                Origin = FeedOriginEnum.Types.FeedOrigin.User
+            };
+
+            // Specify the column name and data type. This is just raw data at this point,
+            // and not yet linked to any particular purpose. The names are used to help us
+            // remember what they are intended for later.
+            feed.Attributes.Add(CreateFeedAttribute("Link Text",
+                FeedAttributeType.String));
+            feed.Attributes.Add(CreateFeedAttribute("Link Final Url",
+                FeedAttributeType.UrlList));
+            feed.Attributes.Add(CreateFeedAttribute("Line 1",
+                FeedAttributeType.String));
+            feed.Attributes.Add(CreateFeedAttribute("Line 2",
+                FeedAttributeType.String));
+
+            FeedServiceClient feedServiceClient = client.GetService(Services.V3.FeedService);
+            FeedOperation operation = new FeedOperation()
+            {
+                Create = feed
+            };
+
+            MutateFeedsResponse response = feedServiceClient.MutateFeeds(
+                customerId.ToString(), new[] {operation});
+            string feedResourceName = response.Results[0].ResourceName;
+            Console.WriteLine($"Created feed with resource name '{feedResourceName}'.");
+
+            // After we create the feed, we need to fetch it so we can determine the
+            // attribute IDs, which will be required when populating feed items.
+            GoogleAdsServiceClient googleAdsService = client.GetService(
+                Services.V3.GoogleAdsService);
+            feed = googleAdsService.Search(
+                customerId.ToString(),
+                $"SELECT feed.attributes FROM feed WHERE feed.resource_name = '{feedResourceName}'"
+            ).First().Feed;
+
+            Hashtable attributeIds = new Hashtable
```

Comment:
Please use a Dictonary<string, string> or variant instead.

Reply:
The problem is that this data structure is holding several different types: Strings for the resource names, long IDs, and a list of longs for the Feed Item IDs.  This is following the style in the golden, and lets us pass one single data structure between the methods.  I agree it's a bit opaque, and I'd prefer something more strongly typed... do you have a suggestion as to a better approach?